package/kedro_viz/models/flowchart.py

```diff
@@ -496,189 +610,278 @@ def is_tracking_node(self):
 
     def is_preview_node(self):
         """Checks if the current node has a preview"""
-        try:
-            is_preview = bool(self.viz_metadata["preview_args"])
-        except (AttributeError, KeyError):
+        if not (self.viz_metadata and self.viz_metadata.get("preview_args", None)):
             return False
-        return is_preview
+
+        return True
 
     def get_preview_args(self):
         """Gets the preview arguments for a dataset"""
-        return self.viz_metadata["preview_args"]
+        return self.viz_metadata.get("preview_args", None)
```

Comment:
It seems that you assume self.viz_metadata is always exist, do you need the first condition?

Reply:
sure, we can add it but `get_preview_args` is only called after checking `is_preview_node`, which checks for self.viz_metadata. should we add it again ?